client/signals/handlers.py

```diff
@@ -0,0 +1,134 @@
+"""
+Contains signals, responsible to sync edX with NodeBB.
+As some related event is occurred in edX the signal is received
+and a relevant request is made to NodeBB write api to make that
+change at NodeBB side too.
+"""
+from django.contrib.auth.models import User
+from django.db.models.signals import post_save, pre_delete
+from django.dispatch import receiver
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from openedx.features.openedx_nodebb_discussion.client.tasks import (
+    task_create_user_on_nodebb, task_sync_user_profile_info_with_nodebb,
+    task_delete_category_from_nodebb, task_delete_user_from_nodebb,
+    task_create_category_on_nodebb, task_join_group_on_nodebb,
+    task_unjoin_group_on_nodebb
+)
+from openedx.features.openedx_nodebb_discussion.models import EdxNodeBBCategory
+from student.models import UserProfile, CourseEnrollment
+
+
+@receiver(post_save, sender=User)
+def create_and_update_user_on_nodebb(sender, instance, created, update_fields, **kwargs):
+    """
+    Creates a new user at Nodebb side when a new user is created at edX side. OR
+    Update the previous one if some relevant changes occurs.
+
+    Args:
+        sender (str): Name of Sender model
+        instance: Newly created entry of Model.
+        created: Flag will be true if new instance created or false if upgraded.
+        update_fields:  Contain frozen_set of the name of fields updated.
+        **kwargs:  All remaining fields.
+    """
+    if created:
+        user_data = {
+            'username': instance.username,
+            'email': instance.email,
+            'joindate': instance.date_joined.strftime('%s')
+        }
+        task_create_user_on_nodebb.delay(**user_data)
+    elif update_fields and 'last_login' not in update_fields:
+        """
+        On login `last_login` field change for ignoring this change we used this check.
+        We are expecting last_login will never changed from django admin panel.
+        """
+        user_data = {
+            'fullname': '{} {}'.format(instance.first_name, instance.last_name)
+        }
+        task_sync_user_profile_info_with_nodebb.delay(username=instance.username, **user_data)
+
+
+@receiver(post_save, sender=UserProfile)
+def sync_user_profile_info_with_nodebb(sender, instance, **kwargs):
```

Comment:
This should be renamed as the update function, I believe.

Reply:
Done